lib/concerns/listable.rb

```diff
@@ -0,0 +1,35 @@
+module Concerns
+  module Listable
+    # Listable class should have @@all class var available
+
+    def list_all_alpha_by_name
+      list_alpha_by_name(all)
+    end
+
+    def list_alpha_by_name(items)
+      list_by_attributes(items, :name)
+    end
+
+    def sort_alpha_by_name(collection)
+      collection.sort_by(&:name)
+    end
+
+    def list_all_by_attributes(*attrs)
+      list_by_attributes(all, *attrs)
+    end
+
+    def list_by_attributes(collection, *attrs)
+      sort_alpha_by_name(collection).each_with_index do |item, i|
+        str = "#{i + 1}. "
+
+        attrs.each do |attr|
+          str += "#{item.send(attr)}"
+          str += " - " unless attr == attrs.last
```

Comment:
I like it. Keep it. But!

This is mutating a string. So it is not Ruby 3.0 compatible (whenever that comes out). The `# frozen_string_literal: true` comment that you see at the top of many Ruby files will make this code break. Because it will enforce that all of your string literals are immutable.

Can you write it in such a way that doesn't mutate a string? (Hint: You can create more strings. You just can't mutate any of them.)